packages/grpc-js/src/server.ts

```diff
@@ -144,9 +144,14 @@ export class Server {
     throw new Error('Not implemented. Use addService() instead');
   }
 
-  addService(
-    service: ServiceDefinition,
-    implementation: UntypedServiceImplementation
+  addService<
+    ImplementationType extends Record<
+      string,
+      UntypedHandleCall
+    > = UntypedServiceImplementation
+  >(
+    service: ServiceDefinition<ImplementationType>,
```

Comment:
I think this might need to be `Partial<ServiceDefinition<ImplementationType>>`. A service definition is allowed to be missing some methods.

Reply:
Don't think `Partial<T>` is needed here as `ServiceDefinition<T>` is defined in such a way that allows the service definition to be missing some methods?

```ts
export interface UntypedServiceImplementation {
  [name: string]: UntypedHandleCall;
}

export type ServiceDefinition<
  ImplementationType = UntypedServiceImplementation
> = {
  readonly [index in keyof ImplementationType]: MethodDefinition<any, any>;
};
```

But I may be wrong... 🤔

Reply:
Oh, right, the problem might actually be the other way around: if the implementation object has methods that aren't part of the service definition, those keys will be considered missing in the service definition.

Reply:
And I think the `Partial<T>` there solves that problem: the inferred `ServiceDefinition<ImplementationType>` type may have extra keys corresponding to implementation object methods that are not part of the service, and partial allows those to be omitted from the service definition object.

Reply:
Will action shortly – just bogged down with other work; sorry!

Reply:
@murgatroid99 – would `addService<ImplementationType extends UntypedServiceImplementation = UntypedServiceImplementation>(...)` work by any chance? Will also check locally how that behaves.

Reply:
So I looked into it, pushed new update – that's the best I can manage at this stage. Seems to be working.

<img width="984" alt="Screenshot 2020-09-02 at 16 06 24" src="https://user-images.githubusercontent.com/4658565/92001203-fcb40e00-ed70-11ea-9265-5ca7dc890b01.png">